src/js/game/shape_definition_manager.js

```diff
@@ -144,6 +144,21 @@ export class ShapeDefinitionManager extends BasicSerializableObject {
         ));
     }
 
+    /**
+     * Generates a definition for stacking the upper definition onto the lower one
+     * @param {ShapeDefinition} definition
+     * @returns {[ShapeDefinition, ShapeDefinition]}
+     */
+    shapeActionUnstack(definition) {
+        const key = "unstack:" + definition.getHash();
+        if (this.operationCache[key]) {
+            return /** @type {[ShapeDefinition, ShapeDefinition]} */ (this.operationCache[key]);
+        }
+
+        const unstacked = definition.cloneAndUnstack();
+        return /** @type {[ShapeDefinition, ShapeDefinition]} */ (this.operationCache[key] = unstacked);
```

Comment:
This is not correct - it sometimes just returns `[ShapeDefinition]`. If you adjust it the way I commented above, this comment is obsolete though.